mqtt/mqtt-bridge/src/persist/mod.rs

```diff
@@ -0,0 +1,49 @@
+use std::error::Error;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures_util::stream::Stream;
+use mqtt3::proto::Publication;
+use parking_lot::Mutex;
+
+mod memory;
+
+/// Persistence used in bridge.
+/// Elements are added, then can be removed once retrieved by the loader
+/// If one attempts to remove added elements before reading via the loader remove returns None
+#[async_trait]
+trait Persist<'a> {
+    type Loader: Stream;
+
+    async fn new(batch_size: usize) -> Self;
+
+    async fn push(&mut self, message: Publication) -> Result<Key, Box<dyn Error>>;
```

Comment:
Is the correct way to return a generic error type?

Reply:
I think it's adding a new type: type Error; 
then return Result<Key, Self::Error>